### PR TITLE
react-assignment

### PR DESCRIPTION
**Pull request by tbarous**

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    res.render('pages/app', {env});
+});
+
+app.get('/favorites', function (req, res) {
+    res.render('pages/app', {env});
+});
+
+app.listen(port, () => {
+    console.log(`Cats app listening on port ${port}`)
+})
```
Comment (ngkamperlo) [inline server.js:26]: Why did you create your own server?

File: src/components/Modal.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+const Close = styled(Times)`
+  position: absolute;
+  right: .5rem;
+  top: .5rem;
+  cursor: pointer;
+  color: white;
+  font-weight: bold;
+  width: 30px;
+  height: 30px;
+  z-index: 99999999999;
```
Comment (ngkamperlo) [inline src/components/Modal.tsx:47]: Why such long number?

File: src/components/Navbar.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+const L = styled(Link)`
+  margin-right: 2rem;
+`;
+
+const Navbar = () => {
+    return (
+        <Nav>
+            <Container>
+                <div style={{display: "flex"}}>
```
Comment (ngkamperlo) [inline src/components/Navbar.tsx:30]: Why did you use `style` at this point? could you have used something else instead?

File: src/pages/Breeds.tsx [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+const CatImage = styled(ImageComponent)`
+  height: 200px;
+  margin-bottom: 1rem;
+`;
+
+const Title = styled(Text)`
+  padding: 1rem;
+`;
+
+interface Props {}
```
Comment (ngkamperlo) [inline src/pages/Breeds.tsx:32]: Is this a leftover or is it needed somewhere?

**Review by ngkamperlo — COMMENTED** [review]
Thank you very much for the effort you put in this assignment :) Apart from the questions below, can you also please answer the following:
1. How come you used typescript instead of plan javascript?
2. Are you aware of other UI libraries ? can you list some of those?
Thanks in advance and again thank you for your time! Looking forward to your responses :)

File: server.js (hunk 1 above)
Comment (tbarous) [reply server.js:26]: It gave me a better control on what pages are rendered on which routes on inital renders. Also, i had more control on the app environment by using a template engine like ejs.

File: src/components/Modal.tsx (hunk 2 above)
Comment (tbarous) [reply src/components/Modal.tsx:47]: Yes this is bad practice. Normally, z-indexes would exist inside the theme for each element or seperately in an object. That way the z-index hierarchy would be more clear. Due to the nature of the project being an assignment some things are not developed in the best possible way.

File: src/components/Navbar.tsx (hunk 3 above)
Comment (tbarous) [reply src/components/Navbar.tsx:30]: :) Nice eye. This is the only style attribute i have used in the app. Yes, styled components could help here.

File: src/pages/Breeds.tsx (hunk 4 above)
Comment (tbarous) [reply src/pages/Breeds.tsx:32]: For future use, probably should have been omitted.

**Comment by tbarous** [comment]
> Thank you very much for the effort you put in this assignment :) Apart from the questions below, can you also please answer the following:
> 
> 1. How come you used typescript instead of plan javascript?
> 2. Are you aware of other UI libraries ? can you list some of those?
>    Thanks in advance and again thank you for your time! Looking forward to your responses :)

Thank you.

Typescript has more features than javascript and is also typed. The UI libraries i have used are:

1. bootstrap, materialize [sass]
2. vuetify [vue]
3. react-bootstrap, material-ui [react]

i used react bootstrap here only for the grid system.

**Pull request closed without merging**